src/domainobjects/swap_position.py

```diff
@@ -2,123 +2,189 @@
 from datetime import datetime, timedelta
 import random
 import string
-import timeit
 import pandas as pd
-import logging
 
 class SwapPosition(Generatable):
-
-    def generate(self, record_count, custom_args):
-        config = self.get_object_config()
+    """ Class to generate swap positions. Generate method will generate a set
+    amount of swap positions. Other generation methods are included where swap
+    positions are the only domain object requiring them.
+
+    Process of generating swap positions dependent on swap contracts, for each
+    of these positions, choose a random number of instruments from those
+    generated prior, then for each position type (start of day, intraday,
+    end of day), generate a record for every date from the user specified
+    start-date until today's date.
+    """
+
+    PURPOSES = ['Outright']
+    POSITION_TYPES = ['S', 'I', 'E']
```

Comment:
why are position types S, I and E and not SOD, Intraday and EOD? I feel like we should use full names to be descriptive and clear. SOD and EOD are accepted abbreviations so they're ok but S, I and E don't feel informative